src/main/scala/mesosphere/marathon/core/task/tracker/impl/TaskTrackerActor.scala

```diff
@@ -29,14 +30,14 @@ object TaskTrackerActor {
   private[impl] case class ForwardTaskOp(deadline: Timestamp, taskId: Task.Id, taskStateOp: TaskStateOp)
 
   /** Describes where and what to send after an update event has been processed by the [[TaskTrackerActor]]. */
-  private[impl] case class Ack(initiator: ActorRef, msg: Any = ()) {
+  private[impl] case class Ack(initiator: ActorRef, stateChange: TaskStateChange) {
     def sendAck(): Unit = {
-      initiator ! msg
+      initiator ! stateChange
```

Comment:
I wonder if a "Failed" should be mapped to Status.Failed. Combined with an ask, the Future with then fail instead of just returning a change that includes the failure. That would be closer to what happened before.

What do you think?